### ISSUE TITLE
Off-grid thumb-release can be missed

### ISSUE THREAD

**Issue by joneit** [openfin/foobars#6]
_This issue created on behalf of @RandomGHUser, from [fin-hypergrid Issue #760](https://github.com/fin-hypergrid/core/issues/760):_

> When dragging the scrollbar, if leaving the grid element and releasing the mouse button, the grid does not detect it, and still believes that the mouse button is still down when hovering backing over the grid (so it will scroll even though the mouse button isn't down).

**Comment by joneit** [comment]
@RandomGHUser, Thank you for your feedback.

The `mouseup` event handler is on the `window` and therefore _generally_ receives the event when the mouse is position anywhere in the window (such as outside a hypergrid's `<canvas>` element).

I stress the word _generally_ because I can think of two possible scenarios that could interfere with receipt (and there may be others):
1. When the mouse button is released over another window element that is listening for `mouseup` _and_ which is calling `stopPropagation`.
2. When the mouse button is released over an `<iframe>` element (which has its own hierarchy and its own `window`).

An app-level solution to **scenario 1** above would be to remove the call to `stopPropagation`. If that is not practical (because, say, an ancestor element between the element in question and the window must not receive the event), you could trigger the event on the window explicitly (from that existing `mouseup` handler):
```js
    ...
    e.stopPropagation(); // leave this in place
    window.dispatchEvent('mouseup', e); // trigger on window
    ...
```
An app-level solution to **scenario 2** above would be to listen for the event from the iframe's `window` and trigger it on the parent window:
```js
window.addEventListener('mouseup', function(e) { top.dispatchEvent('mouseup', e) });
```

**An update to `finbars` can effectively address this issue by checking in its `mousemove` event that the primary mouse button is _still_ down and if it is not, to trigger the `mouseup` handler** (or incorporate its functionality and retire it). (As currently coded, the `mousedown` handler adds the `mousemove` handler and the `mouseup` handler removes it. So it was assumed _(incorrectly as it turns out!)_ that the mouse must be down while the `mousemove` handler is in effect, and for that reason it did not check.)

In the case of Hypergrid, until that update is made and incorporated into a new Hypergrid release, the following hack mitigates the issue somewhat for both scenarios above:
```js
grid.canvas.addEventListener('mouseenter', function(e){
  if ((e.buttons & 1) === 0) { // primary mouse button is up
    window.dispatchEvent(new MouseEvent('mouseup', e)); // (re)trigger, recharacterized as a `mouseup` event
  }
});
```
The reason I say "somewhat" is that although it will cancel the thumb drag on re-entering the grid, the thumb would still be subject to tracking — if, on its way back, the mouse were to move over an element _not subject to_ either scenario above — even though the mouse is now up. (This is due to the defect in the current `mousemove` handler mentioned above.) So better, but not a real solution.

**Comment by joneit** [comment]
Fix pushed to finbars 1.6.1.
